evalml/tests/conftest.py

```diff
@@ -92,6 +92,94 @@ def graphviz():
     return graphviz
 
 
+@pytest.fixture
+def get_test_data_from_configuration():
+    def _get_test_data_from_configuration(input_type, problem_type, column_names=None):
+        X_all = pd.DataFrame(
+            {
+                "all_null": [np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan]
+                * 2,
+                "int_null": [0, 1, 2, np.nan, 4, np.nan, 6] * 2,
+                "age_null": [0, 1, 2, np.nan, 4, np.nan, 6] * 2,
+                "bool_null": [True, None, False, True, False, None, True] * 2,
```

Comment:
I think these will have dtypes of either `float64` for the integer columns or `object` for the boolean column when you create this dataframe.

Do you also need to make sure to handle the case where these columns might have the `Int64` or `boolean` dtype, particularly when testing with the pandas version of the input?

```
"int_null": pd.Series([0, 1, 2, np.nan, 4, np.nan, 6] * 2, dtype="Int64"),
"age_null": pd.Series([0, 1, 2, np.nan, 4, np.nan, 6] * 2, dtype="Int64"),
"bool_null": pd.Series([True, None, False, True, False, None, True] * 2, dtype="boolean"),
```